plugins/module_utils/job.py

```diff
@@ -76,9 +76,47 @@ def _get_job_output(job_id="*", owner="*", job_name="*", dd_name=""):
                 str(rc), str(err)
             )
         )
-    if not out:
-        raise RuntimeError("Failed to retrieve job output. No job output found.")
-    jobs = _parse_jobs(out)
+    jobs = []
+    if out:
+        jobs = _parse_jobs(out)
+    if not jobs:
+        jobs = _job_not_found(job_id, owner, job_name, dd_name)
+
+    return jobs
+
+
+def _job_not_found(job_id, owner, job_name, dd_name):
+    jobs = []
+
+    job = {}
+
+    job["job_id"] = job_id
+    job["job_name"] = job_name
+    job["subsystem"] = "NOTFOUND"
+    job["system"] = "NOTFOUND"
+    job["owner"] = owner
+
+    job["ret_code"] = {}
+    job["ret_code"]["msg"] = "Job not found"
+    job["ret_code"]["code"] = None
+    job["ret_code"]["msg_code"] = "Job not found"
+    job["ret_code"]["msg_txt"] = "The job could not be found"
+
+    job["class"] = ""
+    job["content_type"] = ""
+
+    job["ddnames"] = []
+    dd = {}
+    dd["ddname"] = dd_name
+    dd["record_count"] = "0"
+    dd["id"] = ""
+    dd["stepname"] = "NOTFND"
```

Comment:
Seeing inconsistencies where NOTFND vs NOTFOUND used.

Reply:
https://ibmmainframes.com/references/a21.html  steep name can be 8 characters, max, just like jobname.  So, it could be NOTFOUND.  I'll change to that, so we can see if that helps.